lib/ttnt/storage.rb

```diff
@@ -49,14 +51,32 @@ def write!(section, value)
     private
 
     def filename
-      "#{@repo.workdir}/.ttnt"
+      "#{TTNT.root_dir}/.ttnt"
+    end
+
+    def filename_from_repository_root
+      filename.gsub(@repo.workdir, '')
+    end
+
+    def storage_file_oid
+      tree = @repo.lookup(@sha).tree
+      paths = filename_from_repository_root.split(File::SEPARATOR)
+      dirs, filename = paths[0...-1], paths[-1]
+      dirs.each do |dir|
+        obj = tree[dir]
+        return nil unless obj
+        tree = @repo.lookup(obj[:oid])
+      end
+      obj = tree[filename]
+      return nil unless obj
+      obj[:oid]
     end
 
     def read_storage_content
       if @sha
-        tree = @repo.lookup(@sha).tree
-        if tree['.ttnt']
-          @repo.lookup(@repo.lookup(@sha).tree['.ttnt'][:oid]).content
+        oid = storage_file_oid
+        if oid
```

Comment:
Nice, thank you!
Fixed here: https://github.com/Genki-S/ttnt/commit/3f3190c768b0c7cd92f11785bfde9fbf80ff5c74